packages/flutter/lib/src/widgets/scroll_absolute.dart

```diff
@@ -94,9 +94,16 @@ class ViewportScrollBehavior extends ScrollBehavior2 {
     return null;
   }
 
+  ScrollPhysics _getEffectiveScrollPhysics(BuildContext context, ScrollPhysics physics) {
+    final ScrollPhysics defaultPhysics = getScrollPhysics(getPlatform(context));
+    if (physics is ScrollPhysicsProxy)
+      return physics.applyTo(defaultPhysics);
+    return defaultPhysics;
+  }
```

Comment:
I can make `Scrollable2` only accept `ScrollPhysicsProxy` objects.  I don't think we've found the complete answer to the problem of customizing the behavior, so we're kind of feeling around the problem space at the moment.